src/Selenium2Library/locators/elementfinder.py

```diff
@@ -4,6 +4,25 @@
 from Selenium2Library.utils import escape_xpath_value, events
 
 
+class ParserLocator(object):
+
+    @classmethod
+    def parse_locator(cls, locator):
+        prefix = None
+        criteria = locator
+        if locator.startswith('//') or locator.startswith('(//'):
+            prefix = 'xpath'
+        else:
+            locator_parts = locator.split('=', 1)
+            if len(locator_parts) == 2:
+                prefix = locator_parts[0]
+                criteria = locator_parts[1].strip()
+            elif len(locator_parts) == 1:
+                prefix = 'default'
+                criteria = locator
```

Comment:
Something like this would probably be a bit easier to understand:

```python
if locator.startswith(('//', '(//')):
    return 'xpath', locator
if '=' not in locator:
    return 'default', locator
prefix, locator = locator.split('=', 1)
return prefix, locator.strip()
```